online-inference/tensorizer-isvc/gptj_6b_hf/flask/transformer_flask.py

```diff
@@ -0,0 +1,65 @@
+from typing import List, Dict
+import logging
+import requests
+import numpy as np
+import base64
+import time
+import torch
+from flask import Flask
+from transformers import AutoTokenizer, GPTJForCausalLM
+
+class Transformer(object):
+    def __init__(self):
+        
+        device = "cuda"
+        start = time.time()
+        self.model = GPTJForCausalLM.from_pretrained("/mnt/pvc", torch_dtype=torch.float16).to(device)
```

Comment:
We can save on a bunch of duplicated code by combining the tensorizer and hf inference service scripts.

To do this, we can add a `--tensorizer` command line flag which changes how the model is loaded. Similar to what we do with our stable diffusion inference example: https://github.com/coreweave/kubernetes-cloud/blob/master/online-inference/stable-diffusion/service/service.py

Reply:
I wanted to keep them separate for usability with the trade off of using some duplicated code.

Reply:
@rtalaricw Not a fan of duplicated code. It leads to maintenance issues.

Reply:
I am keeping it separate since we use Flask and KServe. Also we only want the tensorizer example, why would we want HF in Inference Examples? I can just remove HF if we are pushing this to be customer facing.

Reply:
Internally, we wanted to compare HF v/s Tensorizer but if we want to add a README and publish to GitBook, we should keep it clean with only tensorizer.